### ISSUE TITLE
Build failures on older Linuxes

### ISSUE THREAD

**Issue by robdockins** [corsis/clock#35]
```
Clock.hsc: In function ‘main’:
Clock.hsc:142: error: ‘CLOCK_BOOTTIME’ undeclared (first use in this function)
Clock.hsc:142: error: (Each undeclared identifier is reported only once
Clock.hsc:142: error: for each function it appears in.)
```

These clocks were added to Linux relatively recently.  These options should probably be wrapped in `ifdef`s.  To support less-recent Linuxes.

**Comment by ozgurakgun** [comment]
I get essentially the same error but a slightly more verbose error message. Pasting in case it helps.

For info: clock-0.5.1 builds fine on this machine, clock-0.6 doesn't.

Error message is the following.

```
In file included from dist/dist-sandbox-b1d818c0/build/System/Clock_hsc_make.c:1:0:
Clock.hsc: In function ‘main’:
Clock.hsc:142:16: error: ‘CLOCK_BOOTTIME’ undeclared (first use in this function)
/home/ozgur/.tools/ghc/7.10.3/lib/ghc-7.10.3/template-hsc.h:35:10: note: in definition of macro ‘hsc_const’
     if ((x) < 0)                                      \
          ^
Clock.hsc:142:16: note: each undeclared identifier is reported only once for each function it appears in
/home/ozgur/.tools/ghc/7.10.3/lib/ghc-7.10.3/template-hsc.h:35:10: note: in definition of macro ‘hsc_const’
     if ((x) < 0)                                      \
          ^
```

**Comment by jaredmorrow** [comment]
Also getting this failure on Debian 7.

**Comment by alfa07** [comment]
+ Centos 6.7

**Comment by mlongob** [comment]
* RHEL 6.4 - DTS-2

**Comment by cetinsert** [comment]
What does everyone think about #37?

**Comment by mlongob** [comment]
@cetinsert:
I happen to get failures on all 4 options on an older Linux:

```
Clock.hsc:141:16: error: 'CLOCK_MONOTONIC_RAW' undeclared (first use in this function)
---/template-hsc.h:35:10: note: in definition of macro 'hsc_const'
     if ((x) < 0)                                      \
          ^
Clock.hsc:141:16: note: each undeclared identifier is reported only once for each function it appears in
---/template-hsc.h:35:10: note: in definition of macro 'hsc_const'
     if ((x) < 0)                                      \
          ^
Clock.hsc:142:16: error: 'CLOCK_BOOTTIME' undeclared (first use in this function)
---/template-hsc.h:35:10: note: in definition of macro 'hsc_const'
     if ((x) < 0)                                      \
          ^
Clock.hsc:143:16: error: 'CLOCK_MONOTONIC_COARSE' undeclared (first use in this function)
---/template-hsc.h:35:10: note: in definition of macro 'hsc_const'
     if ((x) < 0)                                      \
          ^
Clock.hsc:144:16: error: 'CLOCK_REALTIME_COARSE' undeclared (first use in this function)
---/template-hsc.h:35:10: note: in definition of macro 'hsc_const'
     if ((x) < 0)                                      \
```

I would much rather have this fail at build time. I would just not include the options for the platforms that don't support them. Instead of checking `#if defined (linux_HOST_OS)`, I would check for the existence of each flag. If I find the time today, I can try this out and send you a PR.

**Comment by mlongob** [comment]
I opened #38 for all 4 options

**Comment by robdockins** [comment]
@mlongob Seems like the right fix to me.

**Comment by mlongob** [comment]
@cetinsert Thanks for merging. If you could please make a minor version release, that would be great

**Comment by cetinsert** [comment]
@mlongob I am on it!

**Comment by cetinsert** [comment]
@mlongob released [`clock-0.7.2`](http://hackage.haskell.org/package/clock-0.7.2)!
